### PR TITLE
feat: Add support to continueOn on exit codes. Fixes #6396 

### LINKED ISSUE #6396: Continue on specific exit codes
# Summary

Support configuring specific exit codes with `continueOn`.

# Use Cases

I have a container that can return a custom exit code. I would only like the workflow to continue if the exit code matches that, I don't want it to continue for any failure.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by kriti-sc**
Fixes #6396

### CODE REVIEW

File: workflow/controller/dag.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func (d *dagContext) assessDAGPhase(targetTasks []string, nodes wfv1.Nodes) wfv1
 			// For non-leaf tasks, this is done by setting all of its dependents to allow for their failure or error in
 			// their "depends" clause during their respective "dependencies" to "depends" conversion. See "expandDependency"
 			// in ancestry.go
-			if task := d.GetTask(depName); task.ContinuesOn(branchPhase) {
+			if task := d.GetTask(depName); task.ContinuesOn(branchPhase, d.getTaskNode(depName).Outputs.ExitCode) {
```
Comment (sarabala1979) [inline workflow/controller/dag.go:191]: Can you add test?

File: workflow/controller/steps.go [hunk 2]
```diff
@@ -277,7 +277,7 @@ func (woc *wfOperationCtx) executeStepGroup(ctx context.Context, stepGroup []wfv
 	for _, childNodeID := range node.Children {
 		childNode := woc.wf.Status.Nodes[childNodeID]
 		step := nodeSteps[childNode.Name]
-		if childNode.FailedOrError() && !step.ContinuesOn(childNode.Phase) {
+		if childNode.FailedOrError() && !step.ContinuesOn(childNode.Phase, childNode.Outputs.ExitCode) {
```
Comment (sarabala1979) [inline workflow/controller/steps.go:280]: Can you add test?

**Review by alexec — CHANGES_REQUESTED** [review]
Changes requested by Bala. Please dismiss this review once done.

**Review by alexec — CHANGES_REQUESTED** [review]
Please:

* [ ] Address @simster7 comments.
* [ ] Sync with master.
* [ ] Run `make pre-commit -B` and push changes.
* [ ] Dismiss reviews.

**Comment by tooptoop4** [comment]
where @kriti-sc

**Pull request closed without merging**